Azure Web App deployment fails to build class library project that uses string interpolation

I have an Azure Web App deployed via git that uses the new C# 6 interpolation feature, but the deployment fails. The actual code using string interpolation resides in a class library project, not the web project.

Using https://github.com/KuduApps/Dev14_Net46_Mvc5, I verified that string interpolation code in the web project does not cause a build failure, and it deploys and runs correctly as an Azure Web App.

I forked the above repo - see the "commonproj" branch in https://github.com/jonsagara/Dev14_Net46_Mvc5. I then created a class library project and referenced it from the web project. I moved the string interpolation code into the class library project in a file named CultureHelper.cs. When I try to deploy this as an Azure Web App via git, I get the following error:

```
CultureHelper.cs(19,20): error CS1056: Unexpected character '$' [D:\home\site\repository\Dev14_Net46_Mvc5.Common\Dev14_Net46_Mvc5.Common.csproj]

Failed exitCode=1, command="D:\Windows\Microsoft.NET\Framework\v4.0.30319\MSBuild.exe" "D:\home\site\repository\Dev14_Net46_Mvc5\Dev14_Net46_Mvc5.csproj" /nologo /verbosity:m /t:Build /t:pipelinePreDeployCopyAllFilesToOneFolder /p:_PackageTempDir="D:\local\Temp\f8ad415b-73f5-4460-8aea-9a9643c0590f";AutoParameterizationWebConfigConnectionStrings=false;Configuration=Release /p:SolutionDir="D:\home\site\repository\.\\"

An error has occurred during web site deployment.
```

As requested by [David Ebbo](https://twitter.com/davidebbo/status/633328963192881152), I am filing this issue here. Please let me know if you need more information.

Thank you,

Jon

--- a/Kudu.Contracts/Deployment/LogEntry.cs
+++ b/Kudu.Contracts/Deployment/LogEntry.cs
@@ -40,5 +40,10 @@ namespace Kudu.Core.Deployment
             Message = message;
             Type = type;
         }
+
+        public override string ToString()
+        {
+            return Message;
+        }
     }
 }

--- a/Kudu.Core/Infrastructure/PathUtility.cs
+++ b/Kudu.Core/Infrastructure/PathUtility.cs
@@ -108,8 +108,8 @@ namespace Kudu.Core.Infrastructure
 
         internal static string ResolveMSBuildPath()
         {
-            string windir = SystemEnvironment.GetFolderPath(SystemEnvironment.SpecialFolder.Windows);
-            return Path.Combine(windir, @"Microsoft.NET", "Framework", "v4.0.30319", "MSBuild.exe");
+            string programFiles = SystemEnvironment.GetFolderPath(SystemEnvironment.SpecialFolder.ProgramFilesX86);
+            return Path.Combine(programFiles, @"MSBuild", "14.0", "Bin", "MSBuild.exe");
         }
 
         internal static string ResolveVsTestPath()

--- a/Kudu.FunctionalTests/GitRepositoryManagementTests.cs
+++ b/Kudu.FunctionalTests/GitRepositoryManagementTests.cs
@@ -200,8 +200,8 @@ project = myproject");
                     Assert.Equal(1, results.Count);
                     Assert.Equal(DeployStatus.Failed, results[0].Status);
                     Assert.Null(results[0].LastSuccessEndTime);
-                    KuduAssert.VerifyLogOutput(appManager, results[0].Id, "Warning as Error: The variable 'x' is declared but never used");
-                    Assert.True(deployResult.GitTrace.Contains("Warning as Error: The variable 'x' is declared but never used"));
+                    KuduAssert.VerifyLogOutput(appManager, results[0].Id, "error CS0168: The variable 'x' is declared but never used");
+                    Assert.True(deployResult.GitTrace.Contains("error CS0168: The variable 'x' is declared but never used"));
                     Assert.True(deployResult.GitTrace.Contains("Error - Changes committed to remote repository but deployment to website failed"));
                 });
             }

--- a/Kudu.Services.Web/updateNodeModules.cmd
+++ b/Kudu.Services.Web/updateNodeModules.cmd
@@ -10,7 +10,7 @@ set counter=0
 set /a counter+=1
 echo Attempt %counter% out of %attempts%
 
-cmd /c npm install https://github.com/projectkudu/KuduScript/tarball/38bc1b8b2f863189558a6f51a65c2fc71283df0c
+cmd /c npm install https://github.com/projectkudu/KuduScript/tarball/8038ba6994f16091eca7ab0228a817fd005f53d9
 IF %ERRORLEVEL% NEQ 0 goto error
 
 goto end

--- a/Kudu.TestHarness/TestRepositories.cs
+++ b/Kudu.TestHarness/TestRepositories.cs
@@ -37,7 +37,7 @@ namespace Kudu.TestHarness
             new TestRepositoryInfo("https://github.com/KuduApps/Mvc3AppWithTestProject.git",        "b8c4213"),
             new TestRepositoryInfo("https://github.com/KuduApps/MvcApplicationEFSqlCompact.git",    "c2fa9c0"),
             new TestRepositoryInfo("https://github.com/KuduApps/MvcApplicationWithNuGetAutoRestore","1e269b5"),
-            new TestRepositoryInfo("https://github.com/KuduApps/MVCAppWithLatestNuget.git",         "08e315c"),
+            new TestRepositoryInfo("https://github.com/KuduApps/MVCAppWithLatestNuget.git",         "823836e"),
             new TestRepositoryInfo("https://github.com/KuduApps/MvcAppWithTypeScript.git",          "7319dbb"),
             new TestRepositoryInfo("https://github.com/KuduApps/NodeHelloWorldNoConfig.git",        "d8a15b6"),
             new TestRepositoryInfo("https://github.com/KuduApps/NodeInnerSubDir.git",               "58f93f8"),

